### PR TITLE
Update pull_ipa.py to explicitly pull the binary as well as the .ipa file + bug fix to download

### PR DESCRIPTION
**Pull request by tghosth**
### Added

Changes proposed in this pull request: 
- Add functionality to pull_ipa.py to explicitly pull the binary as well as the .ipa file. Yes I could extract the IPA file manually locally and do that by why not save me a step.
- Extract the unpack IPA code to a separate method so that it can be used independently.
### Fixed

Bug fixes proposed in this pull request:
- The "download" method in remote_operations.py was failing because when the destination path had a space in it, it was getting single quotes added around it. Whilst this is necessary for the source path, when added to the destination path the operation fails to work.

### CODE REVIEW

File: needle/core/device/remote_operations.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def kill_proc(self, procname):
     # ==================================================================================================================
     def download(self, src, dst, recursive=False):
         """Download a file from the device."""
-        src, dst = Utils.escape_path_scp(src), Utils.escape_path_scp(dst)
```
Comment (marco-lancini) [inline needle/core/device/remote_operations.py]: are you sure about this change? at the time I created the `escape_path_scp` just for dealing with scp...

Comment (tghosth) [inline needle/core/device/remote_operations.py]: @marco-lancini did you test this out?

**Comment by marco-lancini** [comment]
Hi @tghosth, are you able to resolve the conflicts so I can merge this?

**Comment by tghosth** [comment]
Hi @marco-lancini, please DON'T merge this yet, I want to test that no issues crept in when I resolved the conflict. I will add another comment here when I have tested this.

Thanks :)

**Comment by tghosth** [comment]
Hi @marco-lancini, I tested it this morning and it seems fine to merge. Thanks :)

**Pull request merged**